sensor/kubernetes/crs/crs.go

```diff
@@ -0,0 +1,341 @@
+	log.Infof("Received CentralID %s", centralHello.GetCentralId())
+
+	// Persisting freshly retrieved service certificates.
+	certsFileMap := centralHello.GetCertBundle()
+	for fileName := range certsFileMap {
+		log.Infof("Received certificate from Central named %s", fileName)
+	}
+	err = persistCertificates(ctx, certsFileMap, k8sClient)
+	if err != nil {
+		return errors.Wrap(err, "persisting certificates")
+	}
+
+	return nil
+}
+
+func useRegistrationSecret(crs *crs.CRS) error {
+	crsTmpDir := os.Getenv(crsTempDirEnvVarName)
+	if crsTmpDir == "" {
+		log.Errorf("environment variable %s must point to a directory suitable for writing sensitive data to", crsTempDirEnvVarName)
+		return errors.Errorf("environment variable %s unset", crsTempDirEnvVarName)
+	}
+
+	// Extract (first) CA certificate from the CRS.
+	var caCert string
+	var err error
+
+	if len(crs.CAs) > 0 {
+		caCert = crs.CAs[0]
+	}
+	if caCert == "" {
+		return errors.New("empty CA in CRS")
+	}
+
+	for fileName, spec := range map[string]struct {
+		setting env.Setting
+		content string
+	}{
+		"ca.pem":   {setting: mtls.CAFilePathSetting, content: caCert},
+		"cert.pem": {setting: mtls.CertFilePathSetting, content: crs.Cert},
+		"key.pem":  {setting: mtls.KeyFilePathSetting, content: crs.Key},
+	} {
+		filePath := filepath.Join(crsTmpDir, fileName)
+		envVar := spec.setting.EnvVar()
+		err = os.WriteFile(filePath, []byte(spec.content), 0600)
+		if err != nil {
+			return errors.Wrapf(err, "writing MTLS material to file %s", filePath)
+		}
+		err = os.Setenv(envVar, filePath)
+		if err != nil {
+			return errors.Wrapf(err, "setting environment variable %s", envVar)
+		}
+		log.Infof("Successfully wrote file %s", filePath)
+	}
+
+	return nil
+}
+
+func persistCertificates(ctx context.Context, certsFileMap map[string]string, k8sClient kubernetes.Interface) error {
+	podName := os.Getenv("POD_NAME")
+	sensorNamespace := pods.GetPodNamespace()
+	secretsClient := k8sClient.CoreV1().Secrets(sensorNamespace)
+
+	typedServiceCerts := protoconv.ConvertFileMapToTypedServiceCertificateSet(certsFileMap)
+	if typedServiceCerts == nil {
+		return errors.New("empty typed service certificates")
+	}
+	ownerRef, err := certrefresh.FetchSensorDeploymentOwnerRef(ctx, podName, sensorNamespace, k8sClient, fetchSensorDeploymentOwnerRefBackoff)
+	if err != nil {
+		return errors.Wrap(err, "fetching sensor deployment owner reference")
+	}
+	repository := securedcluster.NewServiceCertificatesRepo(*ownerRef, sensorNamespace, secretsClient)
+	persistedCertificates, err := repository.EnsureServiceCertificates(ctx, typedServiceCerts)
+	if err != nil {
+		return errors.Wrap(err, "ensuring service certificates are persisted")
+	}
+
+	for _, persistedCert := range persistedCertificates {
+		log.Infof("Persisted certificate and key for service %s", persistedCert.ServiceType.String())
+	}
+	return nil
+}
+
+func getHelmManagedConfig() (*central.HelmManagedConfigInit, error) {
```

Comment:
This and the next function look like code copied from elsewhere? Extract and reuse?